src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -646,6 +770,304 @@ void extendCellMapWithRemoteKeys( VTKMeshGenerator::CellMapType & cellMap )
 }
 
 using CellMapType = std::map< ElementType, std::unordered_map< int, std::vector< vtkIdType > > >;
+/**
+ * @brief Get the tetrahedron node ordering from a vtk polyhedron (42)
+ *
+ * @param cell The VTK cell, type polyhedron (42)
+ * @return std::vector< localIndex > list of nodes
+ */
+std::vector< localIndex > getTetrahedronNodeOrderingFromPolyhedron( vtkCell * const cell )
+{
+  std::vector< localIndex > nodeOrder( 4 );
+
+// Check if the orientation is correct
+  real64 vectorA[3];
+  real64 vectorB[3];
+  real64 vectorC[3];
+  real64 vectorN[3];
+
+  for( localIndex i = 0; i < 3; ++i )
+  {
+    vectorA[i] = cell->GetPoints()->GetPoint( 1 )[i] - cell->GetPoints()->GetPoint( 0 )[i];
+    vectorB[i] = cell->GetPoints()->GetPoint( 2 )[i] - cell->GetPoints()->GetPoint( 0 )[i];
+    vectorC[i] = cell->GetPoints()->GetPoint( 3 )[i] - cell->GetPoints()->GetPoint( 0 )[i];
+  }
+
+  LvArray::tensorOps::crossProduct( vectorN, vectorA, vectorB );
+
+  if( LvArray::tensorOps::AiBi< 3 >( vectorN, vectorC ) > 0 ) // The orientation is correct
+  {
+    nodeOrder[0] = 0;
+    nodeOrder[1] = 1;
+    nodeOrder[2] = 2;
+    nodeOrder[3] = 3;
+  }
+  else // The orientation is incorrect, renumber nodes
+  {
+    nodeOrder[0] = 0;
+    nodeOrder[1] = 2;
+    nodeOrder[2] = 1;
+    nodeOrder[3] = 3;
+  }
+
+  return nodeOrder;
+}
+/**
+ * @brief Get the hexahedron node ordering from a vtk polyhedron (42)
+ *
+ * @param cell The VTK cell, type polyhedron (42)
+ * @return std::vector< localIndex >  list of nodes
+ */
+std::vector< localIndex > getHexahedronNodeOrderingFromPolyhedron( vtkCell * const cell )
+{
+  localIndex iFace;
+  std::vector< localIndex > nodeOrder( 8 );
+  std::map< localIndex, localIndex > G2L;
```

Comment:
I'll have to think about these maps. They have the same problem as `std::set` I commented on earlier, but no such simple solution (we don't have flatmaps). Luckily, this code path is only exercised rarely (for elements that are stored in the VTK dataset using the wrong cell type), so it won't affect performance in the majority of cases. We can postpone optimizing it.

Reply:
At the very least though, I would use `std::unordered_map`. It doesn't seem like iteration order is being relied upon.